File: dayOne/java/levelFour/EncodedString.java
public class EncodedString {

    public static String getEncodedString(String input) {
        String encodedString = "";

        int count = 1;
        for (int index = 0; index < input.length() - 1; index++) {

            if (input.charAt(index) == input.charAt(index + 1))
                count++;

            else {
                System.out.print(input.charAt(index) + " ");
                encodedString += input.charAt(index) + (count == 1 ? "" : count + "");
                count = 1;
            }
        }

        encodedString += input.charAt(input.length() - 1) + (count == 1 ? "" : count + "");

        return encodedString;
    }
}

File: dayOne/python/levelFour/encoded_string.py
def get_encoded_string(string):

    encoded_string = ""
    count = 1

    for index in range(len(string) - 1):

        if string[index] == string[index + 1]:
            count += 1
        else:
            encoded_string += string[index]
            if count > 1:
                encoded_string += str(count)
            count = 1

    encoded_string += string[-1]
    if count > 1:
        encoded_string += str(count)

    return encoded_string